partiql-lang/src/main/kotlin/org/partiql/lang/eval/physical/PhysicalBexprToThunkConverterAsync.kt

```diff
@@ -0,0 +1,362 @@
+package org.partiql.lang.eval.physical
+
+import com.amazon.ionelement.api.BoolElement
+import com.amazon.ionelement.api.MetaContainer
+import org.partiql.annotations.ExperimentalWindowFunctions
+import org.partiql.lang.ast.SourceLocationMeta
+import org.partiql.lang.domains.PartiqlPhysical
+import org.partiql.lang.eval.ExprValue
+import org.partiql.lang.eval.NaturalExprValueComparators
+import org.partiql.lang.eval.ThunkAsync
+import org.partiql.lang.eval.ThunkValueAsync
+import org.partiql.lang.eval.physical.operators.AggregateOperatorFactoryAsync
+import org.partiql.lang.eval.physical.operators.CompiledAggregateFunctionAsync
+import org.partiql.lang.eval.physical.operators.CompiledGroupKeyAsync
+import org.partiql.lang.eval.physical.operators.CompiledSortKeyAsync
+import org.partiql.lang.eval.physical.operators.CompiledWindowFunctionAsync
+import org.partiql.lang.eval.physical.operators.FilterRelationalOperatorFactoryAsync
+import org.partiql.lang.eval.physical.operators.JoinRelationalOperatorFactoryAsync
+import org.partiql.lang.eval.physical.operators.LetRelationalOperatorFactoryAsync
+import org.partiql.lang.eval.physical.operators.LimitRelationalOperatorFactoryAsync
+import org.partiql.lang.eval.physical.operators.OffsetRelationalOperatorFactoryAsync
+import org.partiql.lang.eval.physical.operators.ProjectRelationalOperatorFactoryAsync
+import org.partiql.lang.eval.physical.operators.RelationExpressionAsync
+import org.partiql.lang.eval.physical.operators.RelationalOperatorFactory
+import org.partiql.lang.eval.physical.operators.RelationalOperatorFactoryKey
+import org.partiql.lang.eval.physical.operators.RelationalOperatorKind
+import org.partiql.lang.eval.physical.operators.ScanRelationalOperatorFactoryAsync
+import org.partiql.lang.eval.physical.operators.SortOperatorFactoryAsync
+import org.partiql.lang.eval.physical.operators.UnpivotOperatorFactoryAsync
+import org.partiql.lang.eval.physical.operators.WindowRelationalOperatorFactoryAsync
+import org.partiql.lang.eval.physical.operators.valueExpressionAsync
+import org.partiql.lang.eval.physical.window.createBuiltinWindowFunctionAsync
+import org.partiql.lang.util.toIntExact
+
+/** Converts instances of [PartiqlPhysical.Bexpr] to any [T]. A `suspend` version of the physical plan converter
+ * interface is added since PIG currently does not output async functions.
+ */
```

Comment:
The physical plan to async thunk converter implements this `Converter`, which is the same as the PIG-generated `Converter` but the functions within the interface are all async (i.e. `suspend fun`s). We could consider adding some configurability to PIG to support generating async versions.